File: trainingwork0/答案/提交/支撑材料/code/run_all.py
# %% 自来水厂水质预测与评估 —— 一键运行全部求解脚本
"""
2020 CUMCM A题：自来水厂水质预测与评估
主运行脚本 —— 依次运行数据预处理和四个问题的求解
耗时约 2-3 分钟 (P1最耗时)
"""
import subprocess, sys, os, time

PYTHON = sys.executable
BASE_DIR = os.path.dirname(os.path.abspath(__file__))

scripts = [
    ('数据预处理',       'preprocess.py'),
    ('问题1：因素筛选与预测',   'solve_problem1.py'),
    ('问题2：时滞动态建模',    'solve_problem2.py'),
    ('问题3：混合动态预测',    'solve_problem3.py'),
    ('问题4：水质风险评价',    'solve_problem4.py'),
]

print('=' * 60)
print('A题 自来水厂水质预测与评估 —— 批量求解')
print('=' * 60)

start_all = time.time()
for name, script in scripts:
    print(f'\n{"="*60}')
    print(f'>>> 运行: {name} ({script})')
    print('=' * 60)
    t0 = time.time()
    result = subprocess.run([PYTHON, script],
        capture_output=True, text=True, cwd=BASE_DIR)
    elapsed = time.time() - t0
    if result.returncode == 0:
        print(f'[OK] {name} 完成 (耗时 {elapsed:.1f}s)')
        lines = result.stdout.strip().split('\n')
        for line in lines[-8:]:
            print(f'  {line}')
    else:
        print(f'[FAIL] {name} 失败! (退出码: {result.returncode})')
        print(f'  STDOUT:')
        for line in result.stdout.strip().split('\n')[-15:]:
            print(f'    {line}')
        print(f'  STDERR:')
        for line in result.stderr.strip().split('\n')[-5:]:
            print(f'    {line}')

total = time.time() - start_all
print(f'\n{"="*60}')
print(f'全部完成! 总耗时: {total:.1f}s')
print(f'输出目录: {os.path.join(BASE_DIR, "..", "result")}')
print('=' * 60)

File: 培训作业3/code/run_all.m
%% 蔬菜商品自动定价与补货决策 —— 一键运行全部
% 2023 CUMCM C题
% 依次运行：预处理 → 问题1 → 问题2 → 问题3 → 问题4

%% 初始化
clc;
set(0, 'DefaultFigureVisible', 'off');

base_dir = fileparts(mfilename('fullpath'));
result_dir = fullfile(base_dir, '..', 'result');
if ~exist(result_dir, 'dir'), mkdir(result_dir); end

scripts = {'preprocess', 'solve_problem1', 'solve_problem2', ...
           'solve_problem3', 'solve_problem4'};
labels = {'数据预处理', '问题1：品类分布与关联', ...
    '问题2：销量预测与定价优化', '问题3：选品约束优化', ...
    '问题4：数据采集建议'};

fprintf('=============================================================\n');
fprintf('2023 CUMCM C题：蔬菜商品自动定价与补货决策\n');
fprintf('批量求解 —— 全部四问\n');
fprintf('=============================================================\n');

start_all = tic;

for k = 1:length(scripts)
    fprintf('\n=============================================================\n');
    fprintf('>>> 运行: %s (%s.m)\n', labels{k}, scripts{k});
    fprintf('=============================================================\n');

    try
        run(scripts{k});
        fprintf('[OK] %s 完成\n', labels{k});
    catch ME
        fprintf('[FAIL] %s 失败!\n', labels{k});
        fprintf('  错误: %s\n', ME.message);
        if ~isempty(ME.stack)
            fprintf('  位置: %s (第 %d 行)\n', ME.stack(1).name, ME.stack(1).line);
        end
    end
end

total_time = toc(start_all);
fprintf('\n=============================================================\n');
fprintf('全部完成! 总耗时: %.1fs (%.1f分钟)\n', total_time, total_time/60);
fprintf('输出目录: %s\n', result_dir);
fprintf('=============================================================\n');
